src/Core/MySQL/PacketsProtocolText.cpp

```diff
@@ -52,6 +52,7 @@ void ComFieldList::readPayloadImpl(ReadBuffer & payload)
 {
     // Command byte has been already read from payload.
     readNullTerminated(table, payload);
+    payload.ignore();
```

Comment:
I've tested it locally, and seems like it does not fix the root cause. `payload.ignore()` throws `DB::Exception` with code `ATTEMPT_TO_READ_AFTER_EOF`. ClickHouse server does not write a message about it in log (fixed in #19365) and just sends error message to MySQL client. MySQL client, probably, just ignores error on attempt to load column definitions for autocomplete.

Reply:
The root cause is that `Poco::Net::SecureStreamSocket::receiveBytes(...)` [may return zero bytes](https://github.com/ClickHouse-Extras/poco/blob/a71e0776531861873144e4c4081204ea1a1258fb/NetSSL_OpenSSL/src/SecureSocketImpl.cpp#L521) when reading from blocking socket when connection is not closed. `ReadBufferFromPocoSocket` considers that it's EOF, but actually it isn't and we can continue reading from the socket.

Reply:
Hm, no, actually it happens on unexpectedly closed connection, so Poco code is correct. MySQL client resets connection after receiving responses to `COM_FIELD_LIST` packets for some reason.

Reply:
https://github.com/ClickHouse/ClickHouse/blob/c3fd162cd84a043872e51a77f6d9aa9db76791ed/src/Server/MySQLHandler.cpp#L151
The correct data could not be read, so the link was broken

Reply:
> I've tested it locally, and seems like it does not fix the root cause. `payload.ignore()` throws `DB::Exception` with code `ATTEMPT_TO_READ_AFTER_EOF`. ClickHouse server does not write a message about it in log (fixed in #19365) and just sends error message to MySQL client. MySQL client, probably, just ignores error on attempt to load column definitions for autocomplete.



#19365 
1. which not send error message to the mysql client
2. belongs to the Clickhouse server